### ISSUE TITLE
Sync/link watch data between two sources of the same show

### ISSUE THREAD

**Issue by Duoquadragesimal** [LagradOst/CloudStream-3#1008]
**Guidelines:**
- [x] It **is not** a duplicate issue.
- [x] It **is not** a request to add a website. Write it here instead https://github.com/LagradOst/CloudStream-3/issues/24

**Choose one of the following:**
- New feature

**Describe your request here** *(fill this up)*
- [x] How it works.
A button that would be added in the menu for a show on a provider, e.g. my hero academia on tenshi.moe. It would allow you to sync watch data with another source e.g. AllAnime. Would search for the show, like with the search icon function, and would link them up. so if i watched an episode on tenshi.moe it would be marked as watched on the AllAnime source i linked it with.
- [x] Issues it solves.
Some sources have drawbacks, I started watching My Hero Academia on tenshi.moe which worked on my phone but it can't chromecast. But if I go to a different source to chromecast it, my watch data will be split between the two. This would allow me to use different sources while keeping my watch data accurate
- [x] Why it is needed.
To keep watch data accurate when streaming between different sources - see above.

**Other related info**
-

**Comment by venkat2305** [comment]
An option to change the provider within the video player might help.

**Comment by Duoquadragesimal** [comment]
Yeah maybe? Like you could specify a fallback provider almost, and switch to that whenever you want but the watch data would be on the original provider. That would sove the problem too.

**Comment by Duoquadragesimal** [comment]
Anyone seen this?
